components/ParameterForm.vue

```diff
@@ -118,27 +126,36 @@ const renderAsRadios = (parameter: Parameter) => {
   return parameter.parameterType === ParameterType.Select && optionsAreTerse(parameter);
 };
 
-const submitForm = () => {
-  // Not implemented yet
-};
+const formSubmitting = ref(false);
 
-const largeScreen = ref(true);
-const breakpoint = 992; // CoreUI's "lg" breakpoint
-const setFieldSizes = () => {
-  if (window.innerWidth < breakpoint) {
-    largeScreen.value = false;
-  } else {
-    largeScreen.value = true;
-  }
+const submitForm = async () => {
+  if (!formData.value) {
+    return;
+  };
+
+  formSubmitting.value = true;
+  const response = await $fetch<NewScenarioData>("/api/scenarios", {
+    method: "POST",
+    query: { // Using query instead of body because I couldn't work out how to send a body in the integration test.
+      parameters: formData.value,
+    },
+  }).catch((error: FetchError) => {
+    console.error(error);
+  });
+
+  if (response) {
+    const { runId } = response;
+    navigateToData.value = `/scenarios/${runId}`;
```

Comment:
```suggestion
   await navigateTo(`/scenarios/${runId}`)
```

unless reason for assignment before?

Reply:
This is for testing only, because I couldn't get a spy on navigateTo.

Reply:
I had the comment in the test file - there are two other places it could be, here, and by the data attribute itself.

Reply:
I think our only option besides this is just not to test the form submission as part of component tests. That means we would rely only on e2e tests for the form submission. But since our e2e test doesn't mock the R API, we can't verify that the correct parameters were passed to it by the component.

Reply:
Well, until such time as it can test the result page, I suppose.

Reply:
Emma found this solution: https://github.com/jameel-institute/daedalus-web-app/pull/29